src/cargo/core/compiler/build_runner/mod.rs

```diff
@@ -291,6 +293,14 @@ impl<'a, 'gctx> BuildRunner<'a, 'gctx> {
             }
 
             super::output_depinfo(&mut self, unit)?;
+
+            if self.bcx.build_config.sbom {
+                let sbom = super::build_sbom(&mut self, unit)?;
+                for sbom_output_file in self.sbom_output_files(unit)? {
+                    let outfile = BufWriter::new(paths::create(sbom_output_file)?);
+                    serde_json::to_writer(outfile, &sbom)?;
+                }
```

Comment:
We are outputting the same sbom for each output.

If we're package focused, that makes sense.  If we're create focused, then that doesn't quite work.

Is there anything specific about each root artifact that we'd want to call out?

Reply:
Focusing on the packages makes more sense for SBOMs to me, since that's what's published to crates.io, has CVEs issued against, etc. That said, it's important that we call out if a package is compiled in different ways (different features, etc).

I can't think of anything specific about each root that we'd need to call out.  I don't see an issue with having duplicate contents. If needed, we can always add output-specific information.

Reply:
If we're including the same output, then we're including non-relevant dependencies.  I thought providing focused dependencies was one of the things they were wanting compared to `cargo metadata`.  Granted, this is much easier to trim the unreachable branches but I think its still relevant.

It also feels weird to me to output per artifact the exact same output but I guess that also allows more future evolution.

Reply:
The duplicate contents only occurs when a build unit produces multiple outputs (like `staticlib` + `dylib`). And I don't think it's possible to configure different dependencies for each output type, so the SBOMs should be correctly pruned.

Each root unit within the primary package (say an `example` vs the `main` binary) *can* have different SBOMs. The example, may have `dev-dependencies` that that `main` would not. 

If we're compiling for multiple target triples, those would each also get their own root unit (and their own individual SBOM).

If there's a specific case you're thinking of, let me know and I can add a test to show how it behaves.

Reply:
Ok, I must have misunderstood this to be across units, not within a unit (yup, its within a unit for loop)